app/utils/ATSErrorHandler.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import com.google.inject.Inject
+import models.{BadRequestError, DownstreamClientError, DownstreamServerError, NotFoundError, ServiceError}
+import play.api.Logging
+import play.api.mvc.Result
+import play.api.mvc.Results.{BadGateway, BadRequest, InternalServerError, NotFound}
+
+class ATSErrorHandler @Inject()() extends Logging {
+
+  def errorToResponse(error: ServiceError): Result =
```

Comment:
Missing tests and most likely not needed, as part of your refactor you could just try passing UpstreamErrorResponse as the parameter here and see how the code builds out

Reply:
fixed